src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +47,31 @@
      * Андрей - 600
      * София - 900</p>
      */
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
+
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate localDateTu = LocalDate.parse(dateTo, FORMATTER);
+
+        if (localDateFrom.isAfter(localDateTu)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        StringBuilder result =
+                new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo + "\n");
+        for (String name : names) {
+            int tempSalary = 0;
+            for (String row : data) {
+                String[] dataArray = row.split(" ");
+                LocalDate localWorkDay = LocalDate.parse(dataArray[0], FORMATTER);
+                if (name.equals(dataArray[1])) {
+                    if (!localWorkDay.isBefore(localDateFrom)
+                            && !localWorkDay.isAfter(localDateTu)) {
+                        tempSalary +=
+                                Integer.parseInt(dataArray[2]) * Integer.parseInt(dataArray[3]);
```

Comment:
after removing one `if` statement, all should be fine
```suggestion
                    tempSalary += Integer.parseInt(dataArray[2]) * Integer.parseInt(dataArray[3]);
```